### PR TITLE
Binary attachment uploading

### PR DESCRIPTION
**Pull request by npomfret**

### CODE REVIEW

**Comment by jamesnocentini** [comment]
Thanks for the great work! I'm trying to run the ReactNativeCouchbaseLiteExample app but getting the [following error](http://cl.ly/1Y2N421s2c2Z/sdk_google_phone_x86_64MASTERjamesnocentini04282016104531.png). Does it ring a bell?

File: index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 var { NativeModules } = require('react-native');
 var ReactCBLite = NativeModules.ReactCBLite;
+var FileUpload = require('NativeModules').FileUpload;
```
Comment (jamesnocentini) [inline index.js]: Can you remove this require statement otherwise it throws an error.

**Comment by jamesnocentini** [comment]
This works like a charm @npomfret! I added some docs that I will push once we've merged this PR. There's one change to make (see inline comment) and it would be great to squash the commits into one with the commit message:

```
Changes:
- ReactCBLite.java & ReactCBLite.m: add upload bridge method and SaveAttachmentTask
- index.js: getDocument & getAllDocuments now take an options dictionary as 2nd param, add getAttachmentUri & saveAttachment, refactor makeRequest & _makeRequest & _getFullUrl
```

Thanks again 👍

**Comment by npomfret** [comment]
Merged. https://github.com/couchbaselabs/react-native-couchbase-lite/commit/13499abe8770b9185ab0f26e44113e8a72ebc1e6

**Comment by jamesnocentini** [comment]
Great cheers

**Pull request closed without merging**